### PR TITLE
Added validation for config.shell 

### PR DESCRIPTION
**Pull request by ChrHan**
Added validation for `shell` part of config.

Also revert invalid shell (which for now just looks for valid file) to default.
This should fix issue #1589, and branch is ready to merge.

### CODE REVIEW

File: app/config/init.js [hunk 1]
```diff
@@ -14,6 +15,7 @@ const _syntaxValidation = function (cfg) {
   try {
     return new vm.Script(cfg, {filename: '.hyper.js', displayErrors: true});
   } catch (err) {
+    notify(`Configuration Error! Please fix ~/.hyper.js!`);
```
Comment (CodeTheory) [inline app/config/init.js]: There's no need for this since one line below we show a more in-depth notification.

Comment (ChrHan) [inline app/config/init.js]: Reverted on commit ce4bedd

File: app/config/init.js [hunk 2]
```diff
@@ -39,6 +40,10 @@ const _init = function (cfg) {
     // Ignore undefined values in plugin and localPlugins array Issue #1862
     _cfg.plugins = (_cfg.plugins && _cfg.plugins.filter(Boolean)) || [];
     _cfg.localPlugins = (_cfg.localPlugins && _cfg.localPlugins.filter(Boolean)) || [];
+    if (!fs.existsSync(_cfg.config.shell)) {
```
Comment (ppot) [inline app/config/init.js]: test this before. No need to add fs there. The script will know it if `_cfg.config.shel` exist before the assignation.

**Comment by ChrHan** [comment]
@Stanzilla not sure if I'd be able to pick this PR on the following weeks; you can close it if necessary

**Comment by chabou** [comment]
As discussed this is not the right solution for this problem.

BTW thank you so much for your help and really sorry for answer delays 🙏

**Pull request closed without merging**